packages/frontend/src/components/wallet/Sidebar.js

```diff
@@ -0,0 +1,58 @@
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+
+import CreateCustomName from './CreateCustomName';
+import ExploreApps from './ExploreApps';
+
+const StyledContainer = styled.div`
+    background-color: black;
+    border-radius: 8px;
+    padding-bottom: 30px;
+    margin-bottom: 40px;
+    .dots {
+        margin-top: -30px;
+        display: flex;
+        align-items: center;
+        justify-content: center;
+        .dot {
+            width: 8px;
+            height: 8px;
+            background-color: #D5D4D8;
+            border-radius: 50%;
+            margin: 0 5px;
+            cursor: pointer;
+            
+            &.active {
+                cursor: default;
+                background-color: #8FCDFF;
+            }
+        }
+    }
+`;
+
+export default ({ availableAccounts }) => {
+    const [index, setIndex] = useState(0);
+    useEffect(() => {
+        if (availableAccounts.length > 0) {
+            setIndex(availableAccounts.filter(a => a.length < 64).length > 0 ? 0 : 1);
+        }
+    }, [availableAccounts]);
+    const components = [
+        <ExploreApps />,
+        <CreateCustomName />
+    ];
+    return (
+        <StyledContainer>
+            {components.map((component, i) => {
+                if (i === index) {
+                    return (
+                        <div key={i}>{components[i]}</div>
+                    );
+                }
+            })}
```

Comment:
This is a bit cryptic and I think it could be simplified by not mapping across the available components at all.


We only have two choices here, and we select one of the two based on whether or not there are any non-implicit accounts available -- so I think that this can be an explicit choice  that eliminates a lot of cognitive overhead by removing the array indexes and the meaning behind them.

e.g. something like:

```javascript
export default ({ availableAccounts }) => {
    const [activeComponent, setActiveComponent] = useState(CreateCustomName);

    useEffect(() => {
        if (availableAccounts.length > 0) {
            const numNonImplicitAccounts = availableAccounts.filter(a => a.length < 64).length
            setActiveComponent(numNonImplicitAccounts === 0 ? CreateCustomName : ExploreApps);
        }
    }, [availableAccounts]);

    return (
        <StyledContainer>
            {hasNonImplicitAccounts ? <ExploreApps /> : <CreateCustomName />}
            <div className='dots'>
                <div className={`dot ${activeComponent === CreateCustomName ? 'active' : ''}`} onClick={() => setActiveComponent(CreateCustomName)}></div>
                <div className={`dot ${activeComponent === ExploreApps ? 'active' : ''}`} onClick={() => setActiveComponent(ExploreApps)}></div>
            </div>
        </StyledContainer>
    );
};
```

You might consider it slightly less DRY but overall I think it would be a lot easier to read if we didn't use an array and the index of a component...plus you could explicitly give each div a `key` that matches the type of component being rendered if that's necessary.

What do you think?